src/compiler/transformers/es2015.ts

```diff
@@ -1571,6 +1578,78 @@ namespace ts {
                         break;
                 }
             }
+
+            /**
+             * Lazily creates the way class members will access the class prototype.
+             */
+            function getPrototypeStorageName() {
+                if (prototypeStorageName) {
+                    return prototypeStorageName;
+                }
+
+                const originalPrototypeAccess = createPropertyAccess(getInternalName(node), "prototype");
+
+                // If the class has exactly one non-static member, it'll access prototype members on itself:
+                //   ClassName.prototype.member = ...
+                if (!membersContainMultipleUniqueNames(node.members.filter(classMemberAssignsToPrototype))) {

```

Comment:
Why compute this lazily? You could do this above at around line 1550 and just generate the variable and insert the statement before adding the members. Then you could just pass the variable name to `getClassMemberPrefix` rather than a function.

Also, this approach is fairly inefficient. You filter `node.members`, which enumerates all members and produces a new array, and then enumerate them again inside of `membersContainMultipleUniqueNames` (which exits enumeration early, if possible). It would be more efficient to just check `classMemberAssignsToPrototype` inside of `membersContainMultipleUniqueNames`, since there's only one usage of each.